storage/range.go

```diff
@@ -567,66 +587,62 @@ func (r *Range) endCmd(cmdKey interface{}, args proto.Request, err error, readOn
 // with the command queue or the timestamp cache, as admin commands
 // are not meant to consistently access or modify the underlying data.
 // Admin commands must run on the leader replica.
-func (r *Range) addAdminCmd(ctx context.Context, args proto.Request, reply proto.Response) error {
+func (r *Range) addAdminCmd(ctx context.Context, args proto.Request) (proto.Response, error) {
 	header := args.Header()
 
 	if err := r.checkCmdHeader(header); err != nil {
-		reply.Header().SetGoError(err)
-		return err
+		return nil, err
 	}
 
 	// Admin commands always require the leader lease.
 	if err := r.redirectOnOrAcquireLeaderLease(tracer.FromCtx(ctx), header.Timestamp); err != nil {
-		reply.Header().SetGoError(err)
-		return err
+		return nil, err
 	}
 
-	switch args.(type) {
+	var err error
+	var reply proto.Response
+	switch tArgs := args.(type) {
 	case *proto.AdminSplitRequest:
-		r.AdminSplit(args.(*proto.AdminSplitRequest), reply.(*proto.AdminSplitResponse))
+		var resp proto.AdminSplitResponse
+		resp, err = r.AdminSplit(tArgs)
```

Comment:
There's no reason that `err` has to be set in the outer scope. I would make this 

    resp, err := r.AdminSplit(tArgs)
    return &resp, err

I have some doubts about returning the protos as value types from the individual methods when they're pointers everywhere else. In this case we could do `return r.AdminSplit(tArgs)` if the return types were the same (although maybe that invites trouble as a nil `*proto.AdminSplitResponse` could turn into a non-nil `proto.Response`).